modules/prebid/rulesengine/module.go

```diff
@@ -104,19 +109,31 @@ func (m Module) Shutdown() {
 }
 
 // rebuildTrees returns true if the trees for this account need to be rebuilt; false otherwise
-func rebuildTrees(co *cacheEntry, jsonConfig *json.RawMessage) bool {
-	if !expired(&timeutil.RealTime{}, co.timestamp) {
-		return false
+func rebuildTrees(co *cacheEntry, jsonConfig *json.RawMessage) (bool, error) {
+	updateFrequency, err := jsonparser.GetInt(*jsonConfig, "updatetreefrequencyminutes")
+	if err != nil && err != jsonparser.KeyPathNotFoundError {
+		return false, err
+	} else if err != nil && err == jsonparser.KeyPathNotFoundError {
+		updateFrequency = 0
+	}
+
+	if updateFrequency <= 0 {
+		return false, nil
+	}
+
+	if !expired(&timeutil.RealTime{}, co.timestamp, updateFrequency) {
+		return false, nil
 	}
-	return configChanged(co.hashedConfig, jsonConfig)
+	return configChanged(co.hashedConfig, jsonConfig), nil
 }
 
 // expired returns true if the refresh time has expired; false otherwise
-func expired(t timeutil.Time, ts time.Time) bool {
+func expired(t timeutil.Time, ts time.Time, updateFrequency int64) bool {
 	currentTime := t.Now().UTC()
 
 	delta := currentTime.Sub(ts.UTC())
-	if delta.Seconds() > fiveMinutes.Seconds() {
+	freq := time.Duration(updateFrequency) * time.Minute
+	if delta.Seconds() > freq.Seconds() {
 		return true
 	}
 	return false
```

Comment:
Does this account for an `updateFrequency` of value `0` which will never update?

Reply:
If update frequency is 0 this function will never be executed, because we return right before it:
```golang
if updateFrequency <= 0 {
		return false, nil
}

if !expired(&timeutil.RealTime{}, co.timestamp, updateFrequency) {
	return false, nil
}
```
The `expired` function executes only for non-zero and non-negative update frequency